controlplane/balancer.h

```diff
@@ -133,7 +135,23 @@ class balancer_t : public module_t, common::icp_proto::BalancerService
 	friend class telegraf_t;
 	counter_t<balancer::service_counter_key_t, (size_t)balancer::service_counter::size> service_counters;
 	counter_t<balancer::real_counter_key_t, (size_t)balancer::real_counter::size> real_counters;
+
 	void RealFind(google::protobuf::RpcController* controller, const common::icp_proto::BalancerRealFindRequest* request, common::icp_proto::BalancerRealFindResponse* response, google::protobuf::Closure* done) override;
 	void Real(google::protobuf::RpcController* controller, const ::common::icp_proto::BalancerRealRequest* request, ::common::icp_proto::Empty* response, ::google::protobuf::Closure* done) override;
 	void RealFlush(google::protobuf::RpcController* controller, const ::common::icp_proto::Empty* request, ::common::icp_proto::Empty* response, ::google::protobuf::Closure* done) override;
+
+private:
+	bool reconfigure_wlc();
+	uint32_t calculate_wlc_weight(uint32_t weight, uint32_t connections, uint32_t weight_sum, uint32_t connection_sum, uint32_t wlc_power);
+
+	template<typename Map>
+	std::optional<typename Map::mapped_type::value_type> get_effective_weight(const Map& map, const balancer::real_key_global_t& key) const
+	{
+		auto it = map.find(key);
+		if (it != map.end() && it->second.has_value())
+		{
+			return it->second.value();
+		}
```

Comment:
isn't that equivalent to just 
```suggestion
		if (it != map.end())
		{
			return it->second;
		}
```